wgsl/Makefile

```diff
@@ -28,10 +28,19 @@ img/%.mmd.svg: diagrams/%.mmd ../tools/invoke-mermaid.sh ../tools/mermaid.json
 
 TREESITTER_GRAMMAR_INPUT := grammar/grammar.js
 TREESITTER_PARSER := grammar/build/wgsl.so
-
-# Extract WGSL grammar from the spec, validate it by building a Treesitter parser from it.
-$(TREESITTER_GRAMMAR_INPUT) $(TREESITTER_PARSER): index.bs scanner.cc extract-grammar.py
-	source ../tools/custom-action/dependency-versions.sh && python3 ./extract-grammar.py --spec index.bs --scanner scanner.cc --tree-sitter-dir grammar --flow xb
+# Extract WGSL grammar from the spec
+$(TREESITTER_GRAMMAR_INPUT): index.bs scanner.cc extract-grammar.py
+	source ../tools/custom-action/dependency-versions.sh && python3 ./extract-grammar.py --spec index.bs --scanner scanner.cc --tree-sitter-dir grammar --flow x; \
+
+# Build a Treesitter parser to validate grammar extract and later examples in spec
+$(TREESITTER_PARSER): $(TREESITTER_GRAMMAR_INPUT)
+	source ../tools/custom-action/dependency-versions.sh && python3 ./extract-grammar.py --spec index.bs --scanner scanner.cc --tree-sitter-dir grammar --flow b; \
```

Comment:
I would be tempted to remove the `; \` at the end of this line.  The "if" statement will execute as a second command in the "Make" recipe.

I defer to your choice on this one, because you might be hunting for a race condition , and that race condition might be triggered more easily if the machine does not have to start up a new shell.

Reply:
@dneto0 Oops, forgot those during iteration, thank you very much for pointing out!